### PR TITLE
feat: block new social signup on callback

### PR DESCRIPTION
**Pull request by danielgent**
The next part :+1:

### CODE REVIEW

File: src/authentication-flows/social.tsx [hunk 1]
```diff
@@ -186,6 +188,23 @@ export async function socialAuthCallback({
...
+      // How to actually return JSX here? This function is used in on the AJAX routes...
+      // do we catch this in routes.tsx and then display an error?
+      // presumably we also want this to happen on login2...
+      // this will change the flow on login2... how to guard against this?
+      return ctx.html(
+        <UserNotFound
+          // TODO - do not want to do this! how to rearchitect?
+          vendorSettings={SESAMY_VENDOR_SETTINGS}
+          // oops. just catch exception for now? ugly either way
+          state="1234567"
```
Comment (danielgent) [inline src/authentication-flows/social.tsx:202]: :eyes: the remaining issue with this PR :eyes:

File: src/routes/tsx/routes.tsx [hunk 2]
```diff
@@ -44,23 +43,7 @@ import {
...
-  },
-  loginBackgroundImage: "",
-  checkoutHideSocial: false,
-  supportEmail: "support@sesamy.com",
-  supportUrl: "https://support.sesamy.com",
-  siteUrl: "https://sesamy.com",
-  termsAndConditionsUrl: "https://store.sesamy.com/pages/terms-of-service",
-  manageSubscriptionsUrl: "https://account.sesamy.com/manage-subscriptions",
-};
+import { fetchVendorSettings } from "../../utils/fetchVendorSettings";
```
Comment (danielgent) [inline src/routes/tsx/routes.tsx:46]: a necessary evil :see_no_evil:

File: src/utils/components/UserNotFoundPage.tsx [hunk 3]
```diff
@@ -3,26 +3,27 @@ import { VendorSettings } from "../../types";
...
 };
 
-const UserNotFound: FC<Props> = ({ vendorSettings, state }) => {
+const UserNotFound: FC<Props> = ({ vendorSettings, authParams }) => {
   const linkParams = new URLSearchParams({
-    state,
+    ...authParams,
   });
-  const enterCodeLink = `/u/code?${linkParams}`;
+  const restartFlowLink = `/authorize?${linkParams}`;
```
Comment (danielgent) [inline src/utils/components/UserNotFoundPage.tsx:18]: it's impossible really to test this flow locally, but this should be correct - from here the user has to restart the flow as we don't have a state param that corresponds to a universal auth session here (as the social sign on flow is always restarted from `/authorize`)

File: src/routes/tsx/routes.tsx (hunk 2 above)
Comment (markusahlstrand) [reply src/routes/tsx/routes.tsx:46]: For now :)

File: test/integration/breakit-customizations.spec.ts [hunk 4]
```diff
@@ -101,5 +101,125 @@ test("only allows existing breakit users to progress to the enter code step", as
...
+        state: STATE,
+        client_id: "breakit",
+        nonce: "MnjcTg0ay3xqf3JVqIL05ib.n~~eZcL_",
+        response_type: AuthorizationResponseType.TOKEN_ID_TOKEN,
+        // we DO have an account for this user
+        connection: "demo-social-provider",
+      },
+      connection: "demo-social-provider",
+    }),
+  ).replace("==", "");
```
Comment (markusahlstrand) [inline test/integration/breakit-customizations.spec.ts:215]: This isn't really url-safe.. Think we should switch to using this one: https://oslo.js.org/reference/encoding/base64url

**Review by markusahlstrand — APPROVED** [review]
Looks good. Only a comment about the url encoding

File: test/integration/breakit-customizations.spec.ts (hunk 4 above)
Comment (danielgent) [reply test/integration/breakit-customizations.spec.ts:215]: This is a test though, but you don't mean safe as in crypto hacking etc etc you mean safe as in sanity? :smile:  (so we don't waste time debugging our own base64 encoding function)

I'll try this now on another PR :+1:

**Pull request merged**